src/ray/raylet/node_manager.h

```diff
@@ -0,0 +1,78 @@
+#ifndef NODE_MANAGER_H
+#define NODE_MANAGER_H
+
+#include "ray/raylet/client_connection.h"
+#include "ray/raylet/scheduling_resources.h"
+#include "ray/raylet/scheduling_queue.h"
+#include "ray/raylet/scheduling_policy.h"
+#include "ray/object_manager/object_manager.h"
+#include "ray/raylet/reconstruction_policy.h"
+#include "ray/raylet/task_dependency_manager.h"
+#include "ray/raylet/worker_pool.h"
+
+namespace ray {
+
+class NodeManager : public ClientManager<boost::asio::local::stream_protocol> {
```

Comment:
Rename this to LocalScheduler? That's what it is, right? And "Raylet" is what we used to call the "NodeManager".

Reply:
Hmm, I think this was the way we originally had it but I'm not sure now what the reasoning was for changing it. @atumanov?

Reply:
First, the notion of "local" scheduler is being outdated with this rewrite, because there's no "global" scheduler, which make the term ambiguous and suggestive of the fact there's something other than the "local" scheduler. Second, I think people might simply be used to the term and, naturally, have affinity for calling this component as it was named before. I would encourage to take a step back and think of what this component is actually doing. Which leads to (3) scheduling is only part of the NodeManager's responsibility. In fact, the scheduler is a component inside the NodeManager, periodically invoked to make scheduling decisions. I felt like the NodeManager is more appropriate, as it is more general and descriptive of the fact that this component does more than just scheduling. It performs functions related to the management of the local node. Raylet is really the name of the whole process that includes node management, object management, etc.